### DISCUSSION TITLE
Have I encountered an interesting bug?

### DISCUSSION THREAD

**Discussion by Saeven** [alpinejs/alpine#1688] [7. Bugs (archived)]
I was trying to transform a JSON object into a list, where some list items rely on x-text elements (generating them from template). 

Everything was working fine until I plugged it into its target layout ( which was strange, the changes were minimal).

Hunting for a culprit, I found by commenting out Alpine elements that it was a select list that was causing the problem.  It seems entirely unrelated, which is puzzling.  I don't think I've broken any discrete rules, but let me know if this is my doing.  Seems otherwise, it could be an interesting bug?

Here's a test snippet that identifies the problem I'm running into.

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Test</title>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/alpinejs/3.2.1/cdn.js" defer></script>
</head>
<body>
<div x-data="pageState()">

    <!-- uncomment me to see it break
    <select>
        <template x-for="currency in currencyMap">
            <option
                    :value="currency.code"
                    :selected="currency.code === activeCurrency"
                    x-text="currency.country + ' (' + currency.code + ')'"></option>
        </template>
    </select>
    -->


    <!-- partial:index.partial.html -->
    <div>
        <div x-html="generateList()">

        </div>
        <template x-ref="template-heading">
            <div>
                <span>Heading</span>
                <h1 data-text></h1>
            </div>
        </template>

        <template x-ref="template-item">
            <div>
                <span>Item</span>
                <h3 data-text></h3>
                <p data-description></p>
                <p data-price></p>
            </div>
        </template>
    </div>
</div>

<script>
    function pageState() {
        return {
            activeCurrency: 'USD',

            generateList() {
                let markup = '';
                for (const id of Object.keys(this.items)) {
                    const templateContent = this.$refs["template-" + this.items[id].type];
                    const copy = templateContent.cloneNode(true);

                    let textNode = copy.content.querySelector('[data-text]')
                    textNode.setAttribute('x-text', 'items[' + id + '].title');

                    markup += copy.innerHTML;
                }
                return markup;
            },

            currencyMap: [
                {"code": "USD", "country": "United States"},
                {"code": "CAD", "country": "Canada"}
            ],

            items: {
                "0": {"type": "heading", "title": "Course A"},
                "1": {"type": "item", "title": "Step One", "description": "This is a description for the first item"},
                "2": {"type": "item", "title": "Step Two", "description": "This is a description for the second item"}
            },
        }
    }
</script>
</body>
</html>
```

**How to reproduce:** 

If the select list that relies on the template is commented out, the output is as expected. If you uncomment that select list, it will render to the page, yet breaks output.

No errors are displayed on console.

Thank you in advance for taking a peek.  I know that OSS can be a thankless job - thank you!

**Comment by tvdr** [comment]
An xfor or an xif walks through its parents every child (i think) with initTree()

But doing it in the mutateDom() wrapper,which disables the mutationobserver temporarly.

So in your case, the xfor in the select will initilize the generatelist,which will insert the nodes,but the observer looking for added elements with alpine directive is turned off.

I think i have a few ideas where to fix this, i'll just need to test them which will work

**Comment by calebporzio** [comment]
Yes, you have. I've fixed it in this PR: #1901
